#**Exercise 1:**

There is a 30% chance of rain today, and a 40% chance your ubrella order will arive on time. You are eager to walk in the rain today and cannot do so without either!

What is the probability it will rain AND your umbrella will arrive?

\begin{equation}
P(Rain\space AND\space Umbrella) = {P(Umbrella)\times P(Rain)}
= 0.4 × 0.3 = 0.12
\end{equation}

# **Exercise 2:**
There is a 30% chance of rain today, and a 40% chance your ubrella order will arive on time. You will be able to run errands only if it doesn't rain or your umbrella arrives.
What is probability it will not rain OR your umbrella arrives?

\begin{align}
P(NR \space \text{OR} \space U) = P(NR) + P(U) - P(NR \space AND \space U) \\
P(NR \space \text{OR} \space U) = (1.0 - 0.3) + 0.4 - 0.12 = 0.7 + 0.4 - 0.12 = 0.98
\end{align}



# **Exercise 3:**

There is a 30% chance of rain today, and a 40% chance your ubrella order will arive on time.

However, you found out that if it rains thereis only a 20% chance your ubrella will arive on time. 

What is the probability that it will rain AND your umbrella will arrive on time?

\begin{equation}
P(rain \space AND \space umbrella) = P(Rain) × P(U|Rain) = 0.3 × 0.2 = 0.06
\end{equation}

# **Exercise 4:**
You have 137 passengers booked on a flight from Las Vegas to Dallas. However, it is Las Vegas on a Sunday morning and you estimate each passenger is 40% likely to not show up.

You trying to figure out how many seats to over book, so the plane doesn't fly empty.

How likely is it atleast 50 passengers will not show up?

In [2]:
from scipy.stats import binom
n= 137
p=0.4
p_50_or_more_no_shows=0.0
for x in range(50,138):
  p_50_or_more_no_shows+=binom.pmf(x,n,p)
print(p_50_or_more_no_shows)

0.8220955881474781


# Exercise 5:
You flipped a coin 19 times and got heads 15 times and tails 4 times.
Do you think this coin has any good probability of being fair? Why and why not?

In [4]:
from scipy.stats import beta

a = 15
b = 4

dist= 1.0 - beta.cdf(0.5,a,b)
print(dist)

0.9962310791015625


There is over 99.6% chance that the coin is unfair. Because the with the fair underlying probability rate of 0.5 there is less than 1% chance that this type of outcome is likely to have taken place. 